app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -636,6 +637,14 @@ private void showGeolocationPermissionPrompt() {
         if (geolocationCallback == null) {
             return;
         }
+        if (geoDialog != null && geoDialog.isShowing()) {
+            return;
+        }
+        geoDialog = buildGeoPromptDialog();
```

Comment:
What is the lifecycle of the geoDialog?
A Fragment holds the dialog, and the dialog holds the activity context. Is it possible that recreated activity reuse the fragment and hold the dialog to the activity cause a context leak?

Reply:
Yeah, you're right. I pushed a new version while I'm not 100% sure this will be leak free, can you help take a look at it?